Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+
+            foreach (var branch in branches)
+            {
+                var branchVM = new BranchFormViewModel();
+
+                branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+                branchesVM.Add(branchVM);
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Entities
+        public async Task<IActionResult> ListBranches(int? id)
+        {
+            var applicationDbContext = _context.Branches
+                .Include(b => b.Entity)
+                .Include(b => b.Contact)
+                .Where(m => m.EntityId == id);
+
+            var branches = await applicationDbContext.ToListAsync();
+            var branchesVM = new List<BranchFormViewModel>();
+
+            foreach (var branch in branches)
+            {
+                var branchVM = new BranchFormViewModel();
+
+                branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+                branchesVM.Add(branchVM);
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Branches/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (branch == null)
+            {
+                return NotFound();
+            }
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Create
+        public IActionResult Create()
+        {
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+            return View();
+        }
+
+        // POST: Branches/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(BranchFormViewModel branchVM)
+        {
+            if (ModelState.IsValid)
+            {
+                var branch = _mapper.Map<Branch>(branchVM);
+
+                _context.Add(branch);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            //var branch = await _context.Branches.FindAsync(id);
```

Comment:
Remove useless code.
For searching history in git you have two way git commands or GUI. Most comfortable GUI for you is availiable in VisualStudio. It is a little not intuitive but you can get used to it. Maybe look for some youtube tuts (those are first shots):
https://www.youtube.com/watch?v=7_n-LfAAhiM
https://www.youtube.com/watch?v=Yfz1EZ_YLhE

Using commands (prefered way but it has graduall learning curve - put a lot of effort upfront with small effect)
Have a look into:
1. https://www.youtube.com/watch?v=xJfurQcVYfo
2. https://www.youtube.com/watch?v=rn6v_QgYFnU
3. https://www.youtube.com/watch?v=aOICKP_9xiY
4. https://www.youtube.com/watch?time_continue=4&v=A8Kwocr-K8c&feature=emb_logo

For see spec commit
git show fdf5493
I use git bash and mix output with linux programs like grep, less and so on